tests/test_fpca.py

```diff
@@ -224,8 +227,14 @@ def test_grid_fpca_regularization_fit_result(self):
         fd_data = FDataGrid(np.squeeze(fd_data.data_matrix),
                             np.arange(0.5, 365, 1))
 
-        fpca = FPCAGrid(n_components=n_components, weights=[1] * 365,
-                        regularization_parameter=1)
+        fpca = FPCAGrid(
+            n_components=n_components, weights=[1] * 365,
+            regularization=TikhonovRegularization(
```

Comment:
How does this regularization compare to the previous implemented regularization?

Why the drop in accuracy?

Reply:
The regularization can still be performed using finite differences to approximate derivatives, but for now this has to be explicit (by default it will compute the derivatives using the interpolation). I am not sure if this is desirable though.

The previous approach used only two points to compute each finite difference (accuracy=1) so that they are forward differences, and each time a point was dropped. Now, we use three points to compute finite differences (central differences in the middle points, forward and backwards differences in the extremes, that is, accuracy=2). See https://en.wikipedia.org/wiki/Finite_difference_coefficient for details. Also, the number of points is kept the same. This last property makes it easier to select the points in which the weights of the linear differential operator (which are functions) are evaluated. The previous implementation only allowed constant weights.

The drop in accuracy is then explained because the approximations to the derivative are different.